lexer/lex_test.go

```diff
@@ -851,6 +884,10 @@ func prependOnTestCases(tcase TestCase, tcases []TestCase) []TestCase {
 	return newcases
 }
 
+func sfmt(format string, a ...interface{}) utf16.Str {
+	return Str(fmt.Sprintf(format, a...))
+}
+
 func minusToken() lexer.Tokval {
```

Comment:
There's a lot of tokens. Instead of a function to each token.. why not a simple short constructor helper?
Something like:

```go
func newt(t token.Type, v string) lexer.Tokval {
        return lexer.Tokval{
                Type: t,
                 Value: Str(v),
       }
}

newt(token,Minus, "-")
newt(token.LineTerminator, "\n")
etc
```

Reply:
Well, "newt" goes too far on being small and criptic to my taste =P. The idea implemented now assumes specific functions to each type of token instead of the user having to pass the type (like the minusToken function). Perhaps this generic one could be used on the specific ones (like minusToken)...but I'm not sure if it is such a big deal, lexer.Tokval is not spread on all the tests, they are contained on these token creation functions...but among the creation functions there is this lexer.Tokval initialization that could be considered repetitive...although it does not annoy me that much right now (if it was spread through the tests as it was before...it would annoy me more).

Reply:
For example, using newt there would still be duplication, all tests would have to pass the same token type and the same "-" again and again to represent the "minus token", hence the idea of having specialized functions like minusToken(), it is the option that hides more detail/duplication and that involves less typing that I found until now. 

If the idea is to use newt on these functions...and not on the tests...then Im not sure if it is that big of a duplication. Perhaps as we have more token types there will be. I will hack around the code soon and will take a closer look at this. Thanks =)